File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlHavingClause.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlHavingClause : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.HavingClause;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_HavingClause(this);
    }

    public ISqlExpression Condition { get; set; } = new SqlValue();

    public string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write("HAVING ");
        sql.Write(Condition.ToSql());
        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlHavingClause.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { SqlValue } from './SqlValue';

export class SqlHavingClause implements ISqlExpression {
    SqlType: SqlType = SqlType.HavingClause;
    Span: TextSpan = new TextSpan();
    Condition: ISqlExpression = new SqlValue();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_HavingClause(this);
    }

    ToSql(): string {
        return 'HAVING ' + this.Condition.ToSql();
    }
} 